File: csharp/series/Series.cs
using System;
using System.Collections.Generic;
using System.Linq;

public static class Series
{
    public static string[] Slices(string numbers, int sliceLength)
    {
        if (sliceLength <= 0 || sliceLength > numbers.Length)
        {
            throw new ArgumentException("Invalid slice length", nameof(sliceLength));
        }
        
        return CreateSlices().ToArray();
        
        IEnumerable<string> CreateSlices()
        {
            for (var i = 0; i <= numbers.Length - sliceLength; i++)
            {
                yield return numbers.Substring(i, sliceLength);
            }
        }
    }
}

File: ruby/series/series.rb
class Series
  def initialize(digits)
    @digits = digits
  end

  def slices(series_length)
    raise ArgumentError, 'Invalid length' if
      series_length < 1 || series_length > @digits.length

    @digits.each_char.each_cons(series_length).map(&:join)
  end
end
